llama_index/llms/predibase.py

```diff
@@ -0,0 +1,66 @@
+import os
+from typing import Any, Optional
+
+from llama_index.constants import DEFAULT_CONTEXT_WINDOW, DEFAULT_NUM_OUTPUTS
+from llama_index.llms.base import CompletionResponse, CompletionResponseGen, LLMMetadata
+from llama_index.llms.custom import CustomLLM
+
+
+class PredibaseLLM(CustomLLM):
+    """Predibase LLM"""
+
+    def __init__(
+        self,
+        model_name: str,
+        predibase_api_key: Optional[str] = None,
+        max_new_tokens: int = 256,
+        temperature: float = 0.1,
+        context_window: int = DEFAULT_CONTEXT_WINDOW,
+    ) -> None:
+        self.predibase_api_key = (
+            predibase_api_key
+            if predibase_api_key
+            else os.environ.get("PREDIBASE_API_TOKEN")
+        )
+        self.model_name = model_name
+        self.max_new_tokens = max_new_tokens
+        self.temperature = temperature
+        self.context_window = context_window
```

Comment:
curious: is there a way to automatically get the context window from your LLM implementations?

Reply:
Not currently, though that is something we're considered adding